src/chat.lisp

```diff
@@ -39,30 +42,84 @@
    (raw-data :initarg :raw-data
              :reader get-raw-data)))
 
-
 (defclass private-chat (chat)
-  ((username :initarg :username
-             :reader get-username)
-   (first-name :initarg :first-name
+  ((first-name :initarg :first-name
                :reader get-first-name)
    (last-name :initarg :last-name
-              :reader get-last-name)))
-
+              :reader get-last-name)
+   (bio :initarg :bio
+        :reader get-bio)
+   (has-private-forwards :initarg :has-private-forwards
+                         :reader get-has-private-forwards)))
+
+(defclass base-group (chat)
+  ((linked-chat-id :initarg :linked-chat-id
+                   :reader get-linked-chat-id)
+   (invite-link :initarg :invite-link
+                :reader get-invite-link)
+   (pinned-message :initarg :pinned-message
+                   :reader get-pinned-message)
+   (title :initarg :title
+          :reader get-title)
+   (description :initarg :description
+                :reader get-description)))
+
+(defclass group (base-group)
+  ())
+
+(defclass super-group (base-group)
+  ((join-to-send-messages :initarg :join-to-send-messages
+                          :reader get-join-to-send-messages)
+   (join-by-request :initarg :join-by-request
+                    :reader get-join-by-request)
+   (slow-mode-delay :initarg :slow-mode-delay
+                    :reader get-slow-mode-delay)
+   (sticker-set-name :initarg :sticker-set-name
+                     :reader get-sticker-set-name)
+   (can-set-sticker-set :initarg :can-set-sticker-set
+                        :reader get-can-set-sticker-set)))
+
+(defclass channel (base-group)
+  ())
 
 (defun make-chat (data)
-  (unless (string-equal (getf data :|type|)
-                        "private")
-    (error "Only private chats are supported for now."))
-  
-  (make-instance 'private-chat
-                 :id (getf data :|id|)
-                 :username (getf data :|username|)
+  (when data
+    (let ((type (getf data :|type|)))
+      (apply #'make-instance
+             (cond
+               ((string-equal type "group") 'group)
+               ((string-equal type "supergroup") 'supergroup)
+               ((string-equal type "channel") 'channel)
+               (t 'private-chat))
+             :id (getf data :|id|)
+             :username (getf data :|username|)
+             :has-protected-content (getf data :|has_protected_contents|)
+             :message-auto-delete-time (getf data :|message_auto_delete_time|)
+             :raw-data data
+             (append
+              (when (string-equal type "private")
+                (list
+                 :has-private-forwards (getf data :|has_private_forwards|)
                  :first-name (getf data :|first_name|)
                  :last-name (getf data :|last_name|)
-                 :raw-data data))
+                 :bio (getf data :|bio|)))
+              (unless (string-equal type "private")
```

Comment:
Instead of these `when` and `unless`, try to get supported slot names from the chat class using closer mop.

Here is an example:

```common-lisp
APP/API> (defclass private-chat (cl-telegram-bot/chat:chat)
           ((bio :initarg :bio)))

APP/API> (closer-common-lisp:ensure-finalized (find-class 'private-chat))
#<STANDARD-CLASS APP/API::PRIVATE-CHAT>

APP/API> (closer-common-lisp:class-slots (find-class 'private-chat))
(#<SB-MOP:STANDARD-EFFECTIVE-SLOT-DEFINITION CL-TELEGRAM-BOT/CHAT::ID>
 #<SB-MOP:STANDARD-EFFECTIVE-SLOT-DEFINITION CL-TELEGRAM-BOT/CHAT::RAW-DATA>
 #<SB-MOP:STANDARD-EFFECTIVE-SLOT-DEFINITION APP/API::BIO>)
APP/API> (first *)
#<SB-MOP:STANDARD-EFFECTIVE-SLOT-DEFINITION CL-TELEGRAM-BOT/CHAT::ID>
APP/API> (closer-mop:slot-definition-initargs *)
(:ID)
APP/API> (first *)
:ID
```

This way you will be able to extract from data all fields supported by a class. You only need to transform initarg by downcasing and replacing `-` with `_`.